service/dynamodb/dynamodbattribute/field.go

```diff
@@ -0,0 +1,267 @@
+package dynamodbattribute
+
+import (
+	"reflect"
+	"sort"
+	"strings"
+)
+
+type field struct {
+	tag
+
+	Name        string
+	NameFromTag bool
+
+	Index []int
+	Type  reflect.Type
+}
+
+func fieldByName(fields []field, name string) (field, bool) {
+	for _, f := range fields {
```

Comment:
Will look at that. Used two iterations because an exact match should have priority over an folded match. In this case the fields [`abc`,`Abc`,`ABC`] and searching for `Abc` should return the second element. Since the names are derived from tags and falling back to structure feild name. it is possible to creat a list of marshaled feild names like these.